2.4/root/usr/bin/run-mongod-pet

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Run mongod as part of PetSet.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source ${CONTAINER_SCRIPTS_PATH}/common.sh
+
+function usage() {
+  echo "You must specify the following environment variables:"
+  echo "  MONGODB_USER"
+  echo "  MONGODB_PASSWORD"
+  echo "  MONGODB_DATABASE"
+  echo "  MONGODB_ADMIN_PASSWORD"
+  echo "  MONGODB_KEYFILE_VALUE"
+  echo "  MONGODB_REPLICA_NAME"
+  echo "Optional variables:"
+  echo "  MONGODB_SERVICE_NAME (default: mongodb)"
+  echo "MongoDB settings:"
+  echo "  MONGODB_NOPREALLOC (default: true)"
+  echo "  MONGODB_SMALLFILES (default: true)"
+  echo "  MONGODB_QUIET (default: true)"
+  exit 1
+}
+
+# Make sure env variables don't propagate to mongod process.
+function unset_env_vars() {
+  unset MONGODB_USER MONGODB_PASSWORD MONGODB_DATABASE MONGODB_ADMIN_PASSWORD
+}
+
+function cleanup() {
+  echo "=> Shutting down MongoDB server ..."
```

Comment:
*DISCLAIMER*: this is work-in-progress and all that I say below might change.

## how the scaling will work

### initial deployment

Let's start with a 3-member replica set. `oc new-app template.yaml` will create the replica set.
If any of the 3 pods die, it will be automatically restarted and back to the replica set (without any changes to MongoDB's configuration).

### scaling up

Now the admin decided she wants 5, not 3 members in the replica set. Given that there are appropriate PVs or a dynamic PV provisioner, she runs `oc scale --replicas=5 ...`.
Keep in mind that before scaling up there is some "preparation", or at least making sure the requirements are met.

New pods are created and they connect to the replica set, changing its configuration.

### scaling down

The admin decided to go back to 3 members... so she hits `oc scale --replicas=3 ...`.
At this point we will NOT automatically purge the data in the PVs, instead, that's a manual operation to be performed if she wants to free up the PVs permanently.
Because from the pod's perspective there is no distinction if it terminates because of a crash or a scaling down, it's hostname stays in the replica set config (just like the data is kept).